share/ruby-install/truffleruby/functions.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+platform=$(platform) || return $?
+arch=$(architecture) || return $?
```

Comment:
I'm not sure if `return` is appropriate here (at the top-level of a `source`-d file) but it seems to work fine.
Using `exit` seems to have the same effect for me.
Note that it is needed to `return` or `exit` here, otherwise the script would continue if the platform/architecture is not found (and ultimately fail on download or extraction as `platform` would be an empty string).

Here is the output when the platform is not found:
```
$ bin/ruby-install --no-download truffleruby
!!! unknown platform MyPlatform
zsh: exit 255 
```

Reply:
The #bash channel on Freenode.net seem to be in favor of `return` here, since it's being sourced inside a function and percolates up to a top-level `exit`.

An aside, but they also pointed out that `$?` is the implicit default status when one is omitted from `exit` and `return`. As a separate issue it might be worth considering removing all the `$?`s to reduce clutter.